app/shared/controllers/validators/resolvers/ResolveInteger.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package shared.controllers.validators.resolvers
+
+import cats.data.Validated
+import shared.models.errors.{MtdError, ValueFormatError}
+
+case class ResolveInteger(min: Int = 0, max: Int = 99) extends ResolverSupport {
```

Comment:
Not sure if 99 is a very useful default. It could probably be set to whatever the most common max value is e.g. 99999999999, or Integer.MAX_VALUE maybe (shrug emoji :)

If there's no reasonable default, maybe both args should be required; forces the dev to think about it and supply them.